docs/developer-docs/web-apps/verifiable-credentials/issuer.mdx

```diff
@@ -0,0 +1,164 @@
+---
+keywords: [intermediate, tutorial, identity, credentials, verifiable credentials, issuer]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+# Become an issuer
+
+<MarkdownChipRow labels={["Intermediate", "Tutorial"]} />
+
+## Overview
+
+An **issuer** is a service or app that can issue a verifiable credential (VC) to a user. On ICP, an issuer is an exposed API that recieves calls from an identity provider and does not trigger any workflows itself. To become an issuer, your canister must implement the issuer API as described in [verifiable credential spec](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md).
+
+Issuers receive credentials requests from relying partys through an identity provider such as Internet Identity (II). The end-user must allow or revoke the communication before the credential is issued.
+
+Internet Identity is used as the identity provider in this tutorial.
+
+## Development of the issuer API
+
+The issuer API consists of four endpoints:
+
+1. [`vc_consent_message`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#1-consent-message): Responds with a message of consent to be displayed to the user to receive their consent on issuing the credentials.
+
+2. [`derivation_origin`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#2-derivation-origin): Returns the URL used to derive the user's principal for the issuer. 
+
+3. [`prepare_credential`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#3-prepare-credential): Check the validity of the request and prepare the actual credential requested.
+
+4. [`get_credential endpoint`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#4-get-credential): Issues the actual credential requested by the user.
+
+### Step 1: Obtain a consent message (`vc_consent_message`)
+
+First, the user must send a consent message to the issuer API endpoint `vc_consent_message` through the identity provider. A consent message is a text shown to the user to confirm that they consent to the issuing of a VC.
+
+An example implementation in Rust can be found below:
+
+```rust
+pub fn get_vc_consent_message_en(
+    credential_spec: &CredentialSpec,
```

Comment:
I believe a slightly longer consent message text should be recommended to allowed the user to make an informed decision. 

In the issuer I just built, I used this consent message:

```rust
    let consent_message = format!("<h1>Gitcoin Passport</h1>The relying party is requesting you to share the following credential:<br/><br/>Credential Type: Gitcoin Passport Score<br/>Minimum Score: {min_score}<br/><br/>Sharing the credential DOES NOT mean revealing your exact Passport Score, Ethereum address or other personal information.");
```

https://github.com/kristoferlund/passport-score-issuer/blob/main/packages/issuer_backend/src/service/vc_consent_message.rs

Reply:
Which renders like this:

<img width="464" alt="image" src="https://github.com/dfinity/portal/assets/9698363/aac47bad-a39b-4999-b78f-48dabd990cc1">

Reply:
This is an interesting comment.

I believe we will need a section on best practices on consent messages to keep consistency in the community.

We can work on that after releasing the first draft. This will probably require some discussion within the community.

We'll mention you when we work on that section @kristoferlund . Thanks!